java/component/keyple-core/src/main/java/org/eclipse/keyple/core/plugin/AbstractPlugin.java

```diff
@@ -93,13 +101,51 @@ public final String getName() {
    * @param name of the reader
    * @return the reader
    * @throws KeypleReaderNotFoundException if the wanted reader is not found
+   * @throws IllegalStateException is thrown when plugin is not (or no longer) registered.
    */
   @Override
   public final Reader getReader(String name) {
+    checkStatus();
     Reader reader = readers.get(name);
     if (reader == null) {
       throw new KeypleReaderNotFoundException(name);
     }
     return reader;
   }
+
+  /**
+   * Check if the plugin status is "registered".
+   *
+   * @throws IllegalStateException is thrown when plugin is not (or no longer) registered.
+   */
+  protected void checkStatus() {
+    if (!isRegistered)
+      throw new IllegalStateException(
+          String.format("This plugin, %s, is not registered", getName()));
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void register() {
+    if (isRegistered)
+      throw new IllegalStateException(
+          String.format("This plugin, %s, is already registered", getName()));
+    isRegistered = true;
+    readers.putAll(initNativeReaders());
+    final Collection<Reader> _readers = readers.values();
+    for (Reader seReader : _readers) {
+      seReader.register();
```

Comment:
This action should also be done when observing readers, here:
https://github.com/eclipse/keyple-java/blob/bedfa3221d50697d82101e3d5e417c4b7817d4fb/java/component/keyple-core/src/main/java/org/eclipse/keyple/core/plugin/AbstractThreadedObservablePlugin.java#L183